app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioViewModel.kt

```diff
@@ -241,6 +241,7 @@ class RecordAudioViewModel @Inject constructor(
 
     override fun onCleared() {
         super.onCleared()
+        stopRecording()
```

Comment:
I believe it won't stop recording when you navigate to another screen because in such situation current one is kept in the back stack and its ViewModel is not destroyed. It will only happen when we close the screen and remove it from the back stack, so for instance when navigating back.

Reply:
Good catch! I have moved it to `onDispose` on lifecycle observer in the View Component as we talked.